### PR TITLE
Redesigned empty state

### PR DESCRIPTION
**Pull request by ggirotto**

### CODE REVIEW

File: lib/application/constants/images.dart [hunk 1]
```diff
@@ -17,6 +17,7 @@ enum ImageKey {
   trendingUpArrow,
 
   // illustrations
+  folderIllustration,
```
Comment (ggirotto) [inline lib/application/constants/images.dart:20]: This is the same as `folder` asset, but since `folder` was exported to fit in the tab, it hasn't enough dimensions to be used in the empty state (it gets extremely low quality if stretched to the desired dimensions)

File: lib/application/pages/home/collections/collections_page.dart [hunk 2]
```diff
@@ -91,3 +95,51 @@ class CollectionsContents extends HookWidget {
...
+          strings.collectionsEmptyMessageSegment(currentSegment),
+          style: textTheme.bodyText2?.copyWith(color: theme.neutralSwatch.shade400),
+          textAlign: TextAlign.center,
+        ),
+        context.verticalBox(Spacing.xLarge),
+        ElevatedButton(
+          onPressed: () {
+            final oppositeSegment =
+                currentSegment == CollectionsSegment.explore ? CollectionsSegment.review : CollectionsSegment.explore;
+            onUpdateSegment(oppositeSegment);
```
Comment (ggirotto) [inline lib/application/pages/home/collections/collections_page.dart:137]: Not sure if this is the best solution. Open to suggestions

File: lib/application/constants/images.dart (hunk 1 above)
Comment (matuella) [reply lib/application/constants/images.dart:20]: Then maybe `folderBig`?

Comment (matuella) [reply lib/application/constants/images.dart:20]: Naming it as "illustration" makes others seem to not adhere to this nomenclature

File: lib/application/pages/home/collections/collections_page.dart [hunk 3]
```diff
@@ -91,3 +94,51 @@ class CollectionsContents extends HookWidget {
     return widget.withSymmetricalPadding(context, horizontal: Spacing.medium);
   }
 }
+
+/// [CollectionsPage] empty state, given the current [collectionsVM] state.
```
Comment (matuella) [inline lib/application/pages/home/collections/collections_page.dart:98]:
```suggestion
```

Seems like an irrelevant doc for me, which is incorrectly referring to be related to [collectionsVM] - [_CollectionsContents] is.

File: lib/application/pages/home/collections/collections_page.dart [hunk 4]
```diff
@@ -91,3 +94,51 @@ class CollectionsContents extends HookWidget {
...
+    final theme = useTheme();
+
+    return Column(
+      mainAxisAlignment: MainAxisAlignment.center,
+      children: [
+        Image.asset(
+          images.folderIllustrationAsset,
+          height: dimens.collectionsEmptyStateImageSize,
+          width: dimens.collectionsEmptyStateImageSize,
+          fit: BoxFit.fill,
```
Comment (matuella) [inline lib/application/pages/home/collections/collections_page.dart:117]: fill removes the aspect ratio. You probably want it to be contained

File: lib/application/constants/dimensions.dart [hunk 5]
```diff
@@ -23,6 +23,7 @@ const double smallIconSize = 16;
 // CollectionsPage
 //
 const double collectionsLinearProgressLineWidth = 12;
+const double collectionsEmptyStateImageSize = 96;
```
Comment (matuella) [inline lib/application/constants/dimensions.dart:26]:
```suggestion
const double collectionsEmptyStateSize = 96;
```

File: lib/application/pages/home/collections/collections_page.dart (hunk 2 above)
Comment (matuella) [reply lib/application/pages/home/collections/collections_page.dart:137]: I would remove nested check here, as it can lead to future problems if new tabs are added/updated.

My suggestion is to delegate its responsibility to the caller, something like:

```dart
//...
    child: _CollectionsContents(
      onSegmentSwapRequested: () {
   final nextIndex = segment.index + 1 % availableSegments.length;
   collectionsTabController.animateTo(nextIndex);
},
    ),
//

//...
class _CollectionsContents extends HookWidget {
  const _CollectionsContents({this.onSegmentSwapRequested});
  
  /// {@macro...}
  final VoidCallback onSegmentSwapRequested;
//...
}

//...
class _CollectionsEmptyState extends HookWidget {
  const _CollectionsEmptyState({required this.onSegmentSwapRequested, required this.title, required this.description});
  
  /// {@template}
  /// An empty-state request for a segment swap.
  /// ....
  final VoidCallback onSegmentSwapRequested;
 //...
```

**Review by matuella — CHANGES_REQUESTED** [review]

**Review by matuella — APPROVED** [review]
